eth/peer.go

```diff
@@ -368,7 +372,7 @@ func (p *peer) AsyncSendTransactions(hashes []common.Hash) {
 		for _, hash := range hashes {
 			p.knownTxs.Add(hash)
 		}
-	case <-p.term:
+	default:
```

Comment:
I'm not sure about the `default` change now that the failures drop the peer. Default branches should really only be used for notification channels when it's enough to add one item. Here we're doing data transfers, so a default will cause packets to get lost if the reader is not actively blocked on the read currently. I think we should revert these back to `p.term`.